### PR TITLE
Upstream build fixes from Debian

### LINKED ISSUE #21792: egl: Compile failure as eglextchromium.h replaced by eglext_angle.h in mesa git
### Describe the bug
_Here is a clear and concise description of what the problem is:_

With recent mesa git, its commit [egl: Remove eglextchromium.h and import eglext_angle.h](https://gitlab.freedesktop.org/mesa/mesa/-/commit/f5bb9dd738ace274c97507adea073b6c609469b2) has done what it states and removed the eglextchromium.h header causing  Kodi to fail to compile xbmc/windowing/X11/GLContextEGL.cpp which used that header.

## Expected Behavior
_Here is a clear and concise description of what was expected to happen:_

Compiles.

## Actual Behavior

Does not compile.

## Possible Fix

Quick fix is to replace `#include <EGL/eglextchromium.h>` with `#include <EGL/eglext_angle.h>` in xbmc/windowing/X11/GLContextEGL.cpp but we need to be backwards compatible.

However adding a cmake test fails to "find" eglext_angle.h as the cmake compile test fails due to EGLAPIENTRYP being undefined. The eglext_angle.h is missing an include for eglplatform.h (which xbmc/windowing/X11/GLContextEGL.h does so via including EGL/eglext.h which is probably normal setup). Cannot see a way to get cmake to create a compile test that includes both headers. Trying check_include_files just tests each header individually. (And check_include_file also gives an annoying cmake CMP0075 warning.)

```
diff -ur xbmc/cmake/modules/FindEGL.cmake xbmc-eglextheader/cmake/modules/FindEGL.cmake
--- xbmc/cmake/modules/FindEGL.cmake
+++ xbmc-eglextheader/cmake/modules/FindEGL.cmake
@@ -9,6 +9,7 @@
 # EGL_INCLUDE_DIRS - the EGL include directory
 # EGL_LIBRARIES - the EGL libraries
 # EGL_DEFINITIONS - the EGL definitions
+# HAVE_EGLEXTANGLE - if eglext_angle.h exists else use eglextchromium.h
 #
 # and the following imported targets::
 #
@@ -35,6 +36,11 @@
   set(EGL_LIBRARIES ${EGL_LIBRARY})
   set(EGL_INCLUDE_DIRS ${EGL_INCLUDE_DIR})
   set(EGL_DEFINITIONS -DHAS_EGL=1)
+  include(CheckIncludeFile)
+  check_include_file("EGL/eglext_angle.h" HAVE_EGLEXTANGLE)
+  if(HAVE_EGLEXTANGLE)
+    list(APPEND EGL_DEFINITIONS "-DHAVE_EGLEXTANGLE=1")
+  endif()

   if(NOT TARGET EGL::EGL)
     add_library(EGL::EGL UNKNOWN IMPORTED)
diff -ur xbmc/xbmc/windowing/X11/GLContextEGL.h xbmc-eglextheader/xbmc/windowing/X11/GLContextEGL.h
--- xbmc/xbmc/windowing/X11/GLContextEGL.h
+++ xbmc-eglextheader/xbmc/windowing/X11/GLContextEGL.h
@@ -13,7 +13,11 @@
 #include "threads/CriticalSection.h"

 #include <EGL/eglext.h>
+#ifdef HAVE_EGLEXTANGLE
+#include <EGL/eglext_angle.h>
+#else
 #include <EGL/eglextchromium.h>
+#endif
 #include <X11/Xutil.h>

 class CGLContextEGL : public CGLContext
```

### To Reproduce

1.  Install git mesa
2.  Compile Kodi

### Debuglog

From CMakeError.log with above check_include_file we get;
```
Determining if the include file EGL/eglext_angle.h exists failed with the following output:
Change Dir: CMakeFiles/CMakeTmp

Run Build Command(s):/usr/bin/ninja cmTC_593f3 && [1/2] Building C object CMakeFiles/cmTC_593f3.dir/CheckIncludeFile.c.o
FAILED: CMakeFiles/cmTC_593f3.dir/CheckIncludeFile.c.o
/usr/bin/x86_64-pc-linux-gnu-gcc -DNDEBUG -Wall -o CMakeFiles/cmTC_593f3.dir/CheckIncludeFile.c.o
-c CMakeFiles/CMakeTmp/CheckIncludeFile.c
In file included from CMakeFiles/CMakeTmp/CheckIncludeFile.c:1:
/usr/include/EGL/eglext_angle.h:162:20: error: unknown type name ‘EGLAPIENTRYP’
  162 | typedef EGLBoolean(EGLAPIENTRYP PFNEGLCREATESTREAMPRODUCERD3DTEXTUREANGLEPROC)(EGLDisplay dpy, EGLStreamKHR stream, const EGLAttrib *attrib_list);
      |
In file included from CMakeFiles/CMakeTmp/CheckIncludeFile.c:1:
/usr/include/EGL/eglext_angle.h:163:20: error: unknown type name ‘EGLAPIENTRYP’
....REPEAT FOR ALL LINES WITH EGLAPIENTRYP...
ninja: build stopped: subcommand failed.
```

### Your Environment
Used Operating system: 

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Gentoo
 - Kodi version:19 and 20 affected.

### PR DESCRIPTION
**Pull request by basilgello**
## Description

Upstreaming two Debian patches fixing build and Lintian warnings
and improving quality of Kodi code:

- Get rid of bashisms in kodi*.sh
- Work around Mesa eglchromium.h removal (Fixes #21792)

## Motivation and context

Fixing build with new Mesa 22.3.0 in Debian sid and improving kodi.sh

## How has this been tested?

On Kodi from Debian minions (including myself :P)

## What is the effect on users?

New releases on new systems 

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: cmake/modules/FindEGL.cmake [hunk 1]
```diff
@@ -35,6 +36,11 @@ if(EGL_FOUND)
   set(EGL_LIBRARIES ${EGL_LIBRARY})
   set(EGL_INCLUDE_DIRS ${EGL_INCLUDE_DIR})
   set(EGL_DEFINITIONS -DHAS_EGL=1)
+  include(CheckIncludeFiles)
+  CHECK_INCLUDE_FILES("EGL/egl.h;EGL/eglext.h;EGL/eglext_angle.h" HAVE_EGLEXTANGLE)
```
Comment (wsnipex) [inline cmake/modules/FindEGL.cmake:40]: nitpick: check_include_files

**Review by wsnipex — APPROVED** [review]

**Review by enen92 — APPROVED** [review]
Thanks, runtime tested - works around the mesa issue

**Review by garbear — APPROVED** [review]
Both fixes look good from a code standpoint.

**Comment by basilgello** [comment]
Added one more Debian patch which fix builds on Debian ports architectures (yes, even more rare than s390x ;) )

Proof running Kodi on powerpc64 (ppc64) machine hosted by GCC compiler farm :)
2022-12-30-034742_1280x1024_scrot

As it builds and runs at least on one affected architecture, I think it can be merged as a whole and backported to Nexus.

**Comment by wsnipex** [comment]
so we can build for Amgia 500 now? ;)

**Comment by lrusak** [comment]
> Added one more Debian patch which fix builds on Debian ports architectures (yes, even more rare than s390x ;) )
> 
> 
> 
> Proof running Kodi on powerpc64 (ppc64) machine hosted by GCC compiler farm :)
> 
> 2022-12-30-034742_1280x1024_scrot
> 
> 
> 
> As it builds and runs at least on one affected architecture, I think it can be merged as a whole and backported to Nexus.

You should probably make a separate PR rather than adding more new commits onto this PR as it was already approved in the previous state.

**Comment by basilgello** [comment]
@wsnipex If Debian ports is running on 500, then yes! Surprisingly, all dependencies are there and the only thing was this!
@lrusak OK, will do. Many single commits x backporting them to Nexus = I though re-approval would be easier :)

**Comment by basilgello** [comment]
OK, third commit goes in #22357 !

**Pull request merged**